crates/jstzd/src/task/jstz_node.rs

```diff
@@ -0,0 +1,34 @@
+use super::Task;
+use anyhow::Result;
+use async_trait::async_trait;
+use jstz_node::{config::JstzNodeConfig, run_with_config};
+use tokio::task::JoinHandle;
+
+pub struct JstzNode {
+    handle: JoinHandle<Result<()>>,
+    config: JstzNodeConfig,
+}
+
+#[async_trait]
+impl Task for JstzNode {
+    type Config = JstzNodeConfig;
+
+    async fn spawn(config: Self::Config) -> Result<Self> {
+        let cfg = config.clone();
+        let handle = tokio::spawn(async move { run_with_config(cfg).await });
+        Ok(JstzNode { handle, config })
+    }
+
+    async fn kill(&mut self) -> Result<()> {
+        self.handle.abort();
+        Ok(())
+    }
+
+    async fn health_check(&self) -> Result<bool> {
+        let res = reqwest::get(format!("{}/ping", self.config.endpoint)).await;
+        if res.is_ok_and(|res| res.status().is_success()) {
```

Comment:
What about replacing this `if` with
```rs
Ok(res.is_ok_and(|res| res.status().is_success()))
```
?

Though I think `Ok(res.is_ok())` should be sufficient since 4xx and 5xx status code leads to `Err`.